qutebrowser/completion/completionwidget.py

```diff
@@ -310,6 +312,16 @@ def completion_item_focus(self, which, history=False):
         if not self._active:
             return
 
+        if (
+                config.val.completion.show != 'always'
+                and self.pattern is not None
+                and self.pattern != self._model_pattern
+        ):
+            with debug.log_time(log.completion, 'Set pattern {}'.format(self.pattern)):
+                self.model().set_pattern(self.pattern)
+                self._model_pattern = self.pattern
+                self._maybe_update_geometry()
```

Comment:
You are right. I was adjusting it because I missed that, with `completion.show=auto`, we hide the widget when the user starts typing again. I was keeping it visible, so it needed adjustment if the user hit Tab again and fewer results needed to be displayed. Now it's hidden again when the pattern changes.